Matrix.py

```diff
@@ -0,0 +1,175 @@
+        return matrix_in_lines
+
+    # unity method creates a matrix of zeros with a diagonal of ones (unity is the Matrix's size)
+    @staticmethod
+    def unity(unit):
+        major_list = [[1 if i == index else 0 for i in range(unit)] for index in range(unit)]
+        sub_list = [tuple(lt) for lt in major_list]
+        # converts list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    # ones method returns a Matrix full of ones which sizes determined by "unit"
+    @staticmethod
+    def ones(unit):
+        sub_list = [tuple(lt) for lt in [([1] * unit) for i in range(unit)]]
+        # convert list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    def __add__(self, other):
+        """add a matrix to another one
+            args: self.data (tuple)- A matrix
+                other(tuple) -A matrix
+
+            returns: return the new matrix (tuple)
+
+             raises: syntaxerror if "other" isn't a matrix"""
+        if type(other) is Matrix:
+            answer = []
+            for tup in range(len(self.data)):
+                answer.append([])
+                for num in range(len(self.data)):
+                    answer[tup].append(self.data[tup][num] + other[tup][num])
+            sub_list = [tuple(ls) for ls in answer]
+            # convert answer to type Matrix. ls are the inside lists
+            return Matrix((tuple(sub_list)))
+        else:
+            raise SyntaxError("only matrix's can be added together")
+
+    def __sub__(self, other):
+        """subtracting a matrix from another
+                    args: self.data (tuple)- A matrix
+                        other(tuple) -A matrix
+
+                    returns: return the new matrix (tuple)
+
+                     raises: syntaxerror if "other" isn't a matrix"""
+        if type(other) is Matrix:
+            answer = []
+            for tup in range(len(self.data)):
+                answer.append([])
+                for num in range(len(self.data)):
+                    answer[tup].append(self.data[tup][num] - other[tup][num])
+            sub_list = [tuple(ls) for ls in answer]
+            # converts list of lists to type Matrix. ls are the inside lists
+            return Matrix((tuple(sub_list)))
+        else:
+            raise SyntaxError("only matrix's subscription")
+
+    def __truediv__(self, other):
+        """divides a matrix by scalar
+                    args: self.data (tuple)- A matrix
+                        other(int/float) -A scalar
+
+                    returns: return the new matrix (tuple)
+
+                     raises: syntaxerror if "other" isn't a scalar"""
+        if type(other) is float or int:  # the program doesnt support division by other Matrix
+            return self.scalar_mul(1 / other)
```

Comment:
use regular mul

Reply:
i used it to decide which mul method will be used (scalar/matrices)

Reply:
why?

Reply:
regular mul = *
You defined a mul function... now use it

Reply:
?